workshops/views.py

```diff
@@ -683,126 +686,29 @@ def form_valid(self, form):
-        else:
-            event = None
-    else:
-        event = None
-
-    # PersonAwardForm
-    award_form = PersonAwardForm(prefix='award', initial=dict_without_Nones(
-        awarded=datetime.date.today(),
-        person=person,
-        badge=badge,
-        event=event,
-    ))
-
-    # Determine which form was sent (if any)
-    if request.method == 'POST' and 'award-badge' in request.POST:
-        award_form = PersonAwardForm(request.POST, prefix='award')
-
-        if award_form.is_valid():
-            award = award_form.save()
-
-            messages.success(
-                request,
-                '{person} was awarded {badge} badge.'.format(
-                    person=str(person),
-                    badge=award.badge.title,
-                ),
-                extra_tags='awards',
-            )
-
-            return redirect_with_next_support(
-                request, '{}#awards'.format(request.path))
-
-        else:
-            messages.error(request, 'Fix errors in the award form.',
-                           extra_tags='awards')
-
-    elif request.method == 'POST' and 'task-role' in request.POST:
-        task_form = PersonTaskForm(request.POST, prefix='task')
-
-        if task_form.is_valid():
-            task = task_form.save()
-
-            messages.success(
-                request,
-                '{person} was added a role {role} during {event} event.'
-                .format(
-                    person=str(person),
-                    role=task.role.name,
-                    event=task.event.slug,
-                ),
-                extra_tags='tasks',
-            )
-
-            return redirect('{}#tasks'.format(request.path))
-
-        else:
-            messages.error(request, 'Fix errors in the task form.',
-                           extra_tags='tasks')
-
-    elif request.method == 'POST':
-        person_form = PersonForm(request.POST, prefix='person',
-                                 instance=person)
-        if person_form.is_valid():
-            lessons = person_form.cleaned_data['lessons']
-
-            # remove existing Qualifications for user
-            Qualification.objects.filter(person=person).delete()
-
-            # add new Qualifications
-            for lesson in lessons:
-                Qualification.objects.create(person=person, lesson=lesson)
-
-            # don't save related lessons
-            del person_form.cleaned_data['lessons']
-
-            person = person_form.save()
-
-            messages.success(
-                request,
-                '{name} was updated successfully.'.format(
-                    name=str(person),
-                ),
-            )
-
-            return redirect(person)
+class PersonUpdate(OnlyForAdminsMixin, UserPassesTestMixin,
+                   AMYUpdateView):
+    model = Person
+    form_class = PersonForm
+    pk_url_kwarg = 'person_id'
+    template_name = 'workshops/person_edit_form.html'
```

Comment:
So, the current permission logic is:
- A user can access the `person_edit` view if he is an admin user and has the `workshops.edit_person` permission or he/she is the person being edited.
- If a person satisfies the above criteria, he can access the view. There is another layer of permission handling within the templates.
    - If the person has the `workshops.edit_award` or `workshops.edit_task` permissions, the respective award and task add forms are rendered. Else, he can only view the list of awards and tasks.

The explicit permissions are handled within the templates. Am I missing something?